### PR TITLE
Fix issues #1920 and #1787

### PR DESCRIPTION
**Pull request by tamis-laan**
Fixed the following issues:
https://github.com/qtile/qtile/issues/1920 
https://github.com/qtile/qtile/issues/1787

### CODE REVIEW

File: libqtile/scratchpad.py [hunk 1]
```diff
@@ -184,18 +184,25 @@ def show(self):
...
+            height = int(screen.dheight * self.height)
+            # Set the geometry
+            win.x = x
+            win.y = y
+            win.float_x = x
+            win.float_y = y
+            win.width = width
+            win.height = height
+            # Configure the new geometry
+            win._reconfigure_floating()
```
Comment (ramnes) [inline libqtile/scratchpad.py:204]: This line here is the only real change of the whole diff, right? Can we remove all the noise?

**Comment by ramnes** [comment]
Thanks for the PR! Please see my comment on the diff. How is that helping for #1787?

File: libqtile/scratchpad.py (hunk 1 above)
Comment (tamis-laan) [reply libqtile/scratchpad.py:204]: Sure the only required part is `win._reconfigure_floating()`, I will quickly revert the rest.

**Comment by tamis-laan** [comment]
> Thanks for the PR! Please see my comment on the diff. How is that helping for #1787?
When you move another group to the current screen then the that group is first drawn and then the hooks are called. 

The problem is that in my hook I'm hiding the bar, however because the group is drawn before hiding the bar there a gap will appear where the bar was. 

By first executing the hooks and then drawing the group this problem is resolved. It does result in some duplicate code which is a bit ugly though.

File: libqtile/config.py [hunk 2]
```diff
@@ -348,6 +348,7 @@ def set_group(self, new_group, save_prev=True):
             g2 = new_group
             s2 = new_group.screen
 
+
```
Comment (ramnes) [inline libqtile/config.py:351]: This is going to break the CI I guess! :)

**Comment by ramnes** [comment]
Thanks!

**Pull request merged**